lib-multisrc/mmlook/src/eu/kanade/tachiyomi/multisrc/mmlook/MMLook.kt

```diff
@@ -0,0 +1,200 @@
+package eu.kanade.tachiyomi.multisrc.mmlook
+
+import eu.kanade.tachiyomi.lib.unpacker.Unpacker
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.parseAs
+import okhttp3.FormBody
+import okhttp3.Request
+import okhttp3.Response
+
+// Rumanhua legacy preference:
+// const val APP_CUSTOMIZATION_URL = "APP_CUSTOMIZATION_URL"
+
+/** 漫漫看 */
+open class MMLook(
+    override val name: String,
+    override val baseUrl: String,
+    private val desktopUrl: String,
+    private val useLegacyMangaUrl: Boolean,
+) : HttpSource() {
+    override val lang: String get() = "zh"
+    override val supportsLatest: Boolean get() = true
+
+    override val client = super.client.newBuilder()
+        .followRedirects(false)
+        .hostnameVerifier { _, _ -> true }
+        .build()
+
+    private fun String.certificateWorkaround() = replace("https:", "http:")
+
+    private fun SManga.formatUrl() = apply { if (useLegacyMangaUrl) url = "/$url/" }
+
+    private fun rankingRequest(id: String) = GET("$desktopUrl/rank/$id", headers)
+
+    override fun popularMangaRequest(page: Int) = rankingRequest("1")
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val entries = response.asJsoup().select(".likedata").map { element ->
+            SManga.create().apply {
+                url = element.select("a").attr("href").mustRemoveSurrounding("/", "/")
+                title = element.selectFirst(".le-t")!!.text()
+                author = element.selectFirst(".likeinfo > p")!!.text()
+                    .mustRemoveSurrounding("作者：", "")
+                description = element.selectFirst(".le-j")!!.text()
+                thumbnail_url = element.selectFirst("img")!!.attr("data-src")
```

Comment:
anything apart from url and title should not be mandatory, so extension still works if non-essential data is not available

Reply:
I don't think these websites will deliberately change the DOM structure of just some minor fields. Their code seems to be quite old and not self-developed. They either keep it unchanged or change it all.